okhttp-clients/src/test/java/com/palantir/remoting3/okhttp/ExponentialBackoffTest.java

```diff
@@ -54,4 +74,63 @@ public void testRetriesCorrectNumberOfTimesAndFindsRandomBackoffWithInExponentia
 
         assertThat(backoff.nextBackoff()).isEmpty();
     }
+
+    @Test
+    public void testBackingOff() throws ExecutionException, InterruptedException {
+        int numThreads = 160;
+        int numRequestsPerSecond = 40;
+        int numRetries = 3;
+        ListeningExecutorService executorService = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(numThreads,
+                new ThreadFactoryBuilder().setNameFormat("test-name-%d").build()));
+        RateLimiter rateLimiter = RateLimiter.create(numRequestsPerSecond);
+
+        ConcurrencyLimiter limiter = new ConcurrencyLimiters().limiter("");
+        Meter meter = new Meter();
+        Histogram backoffHistogram = new Histogram(new ExponentiallyDecayingReservoir());
+
+        List<ListenableFuture<?>> futures = IntStream.range(0, numThreads).mapToObj(x -> executorService.submit(new Runnable() {
+            ExponentialBackoff backoff;
+            int backoffIndex = 0;
+
+            @Override
+            public void run() {
+                for (int i = 0; i < 1001;) {
+                    Limiter.Listener listener = Futures.getUnchecked(limiter.acquire());
+                    //System.out.println(i);
```

Comment:
meep

Reply:
I'm going to rewrite these tests into a kinda contract test thing.